src/services/Http.hh

```diff
@@ -0,0 +1,155 @@
+#ifndef RA_SERVICES_HTTP_H
+#define RA_SERVICES_HTTP_H
+#pragma once
+
+#include <functional>
+
+namespace ra {
+namespace services {
+
+class Http
+{
+public:
+    class Response
+    {
+    public:
+        explicit Response(int nStatusCode, const std::string& sResponse) noexcept
+            : m_nStatusCode(nStatusCode), m_sResponse(sResponse)
+        {
+        }
+
+        Response(const Response&) noexcept = default;
+        Response(Response&&) noexcept = default;
+
+        /// <summary>
+        /// Gets the HTTP status code returned from the server.
+        /// </summary>
+        unsigned int StatusCode() const { return m_nStatusCode; }
+
+        /// <summary>
+        /// Gets the content returned from the server.
+        /// </summary>
+        const std::string& Content() const { return m_sResponse; }
+
+    private:
+        unsigned int m_nStatusCode;
+        std::string m_sResponse;
+    };
+
+    class Request
+    {
+    public:
+        explicit Request(const std::string& sUrl) noexcept
+        {
+            auto nIndex = sUrl.find('?');
+            if (nIndex == std::string::npos)
+            {
+                m_sUrl = sUrl;
+            }
+            else
+            {
+                m_sUrl.assign(sUrl, 0, nIndex);
+                m_sQueryString.assign(sUrl, nIndex + 1, std::string::npos);
+            }
+        }
+
+        Request(const Request&) = default;
```

Comment:
You should default and/or delete all of them if you're going to do this. It'll also make it clear what's to be used so we don't have to guess what's implicitly deleted. If you want copying, moving can't be deleted.